### PR TITLE
Book page view

### PR DESCRIPTION
**Pull request by petarLits**

### CODE REVIEW

**Comment by lecicdjuro** [comment]
Bad title

**Review by UrosLITS — APPROVED** [review]

File: lib/book/book_page/new_book_page.dart [hunk 1]
```diff
@@ -52,12 +53,12 @@ class _NewBookPageState extends State<NewBookPage> {
             backgroundColor: AppColors.errorSnackBar,
           );
           ScaffoldMessenger.of(context).showSnackBar(snackBar);
-        } else if (state is BookPageImageAdded) {
+        } else if (state is AddBookPageImageState) {
           image = state.image;
           decodedImage = await decodeImageFromList(image!.readAsBytesSync());
```
Comment (milicaLits) [inline lib/book/book_page/new_book_page.dart]: Maybe put this function in event handler.

File: lib/book/book_page/book_page_view.dart [hunk 2]
```diff
@@ -5,66 +5,243 @@ import 'package:book/book/bloc/book_page_event.dart';
...
+                          ),
+                        ],
+                      ),
+                    ),
+                  )
+                : null,
+          );
+        } else {
+          return Center(
+            child: CircularProgressIndicator(),
```
Comment (milicaLits) [inline lib/book/book_page/book_page_view.dart:146]: I don't think you need Loading and Loaded State if you you build ProgressIndicator when the state is other than DisplayBookPage State.

File: lib/book/bloc/book_page_state.dart [hunk 3]
```diff
@@ -15,9 +16,10 @@ class InitialState extends BookPageState {
 }
 
 class AddBookPageImageState extends BookPageState {
-  AddBookPageImageState({required this.image});
+  AddBookPageImageState({required this.image, required this.decodedImage});
```
Comment (milicaLits) [inline lib/book/bloc/book_page_state.dart]: format?

**Review by milicaLits — APPROVED** [review]

File: lib/book/bloc/book_page_bloc.dart [hunk 4]
```diff
@@ -23,7 +25,9 @@ class BookPageBloc extends Bloc<BookPageEvent, BookPageState> {
 
   Future<void> _onAddBookPageImage(
       AddBookPageImageEvent event, Emitter<BookPageState> emit) async {
-    emit(AddBookPageImageState(image: event.image));
+    ui.Image decodedImage;
+    decodedImage = await decodeImageFromList(event.image!.readAsBytesSync());
+    emit(AddBookPageImageState(image: event.image, decodedImage: decodedImage));
   }
```
Comment (UrosLITS) [inline lib/book/bloc/book_page_bloc.dart]: what if this call fails?

**Review by UrosLITS — CHANGES_REQUESTED** [review]

**Review by UrosLITS — APPROVED** [review]

**Pull request merged**